File: Selenium/Project/Activity4.py
#Logging into the site
#Goal: Open the site and login with the credentials provided


# Import webdriver from selenium
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service = service) as driver:
  
  # Open the browser to the URL
  driver.get("https://alchemy.hguy.co/crm")

  # Read the title of the website and verify the text
  print("Home page title:"+driver.title)
  
  # Find the username field and enter the username
  driver.find_element(By.XPATH,"//input[@id='user_name']").send_keys("admin")
  
  # Find the password field and enter the password
  driver.find_element(By.XPATH,"//input[@id='username_password']").send_keys("pa$$w0rd")
  
  # Find the login button and click it
  driver.find_element(By.XPATH,"//input[@name='Login']").click()
  
  #Verify that the homepage has opened.
  expectedTitle="SuiteCRM";
  print("Title of the WebPage is:"+driver.title)
  if driver.title is not None and expectedTitle in driver.title:
    print("Home page is opened")
  else:
    print("Home page could not open.")
    
  # Close the browser
  driver.close()

File: Selenium/Project/Activity4.java
/*Logging into the site
        Goal: Open the site and login with the credentials provided*/

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.testng.Assert;
import org.testng.annotations.AfterMethod;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

public class Activity4 {
    // Declare the WebDriver object
    private WebDriver driver;

    @BeforeMethod
    public void setupMethod() {
        // Set up the Firefox driver
        WebDriverManager.firefoxdriver().setup();
        //Create a new instance of the Firefox driver
        driver = new FirefoxDriver();
        //Open browser and navigate to url
        driver.get("https://alchemy.hguy.co/crm");
    }

    @Test
    public void loggingIntoSite() {
        // Find the username field and enter the username
        driver.findElement(By.xpath("//input[@id='user_name']")).sendKeys("admin");
        // Find the password field and enter the password
        driver.findElement(By.xpath("//input[@id='username_password']")).sendKeys("pa$$w0rd");
        // Find the login button and click it
        driver.findElement(By.xpath("//input[@name='Login']")).click();
        String expectedTitle="SuiteCRM";
        System.out.println("Title of the WebPage is:"+driver.getTitle());
        if(driver.getTitle() != null && driver.getTitle().contains(expectedTitle)){
            System.out.println("Home page is opened");
        }
        else{
            System.out.println("Home page could not open.");
        }
    }

    @AfterMethod
    public void afterMethod() {
        //Close the browser
        driver.close();
    }

}
